__init__.py

```diff
@@ -43,28 +46,51 @@ def __init__(self):
 
         self.nato_dict = None
 
+        self.mycroft_ready = False
+        self.pair_dialog_lock = Lock()
+        self.pairing_dialog = 'pairing.paired'
+
     def initialize(self):
-        # TODO:18.02 - use decorator
-        intent = IntentBuilder("PairingIntent") \
-            .require("PairingKeyword").require("DeviceKeyword").build()
-        self.register_intent(intent, self.handle_pairing)
         self.add_event("mycroft.not.paired", self.not_paired)
         self.nato_dict = self.translate_namedvalues('codes')
 
+        # If the device isn't paired catch mycroft.ready to report
+        # that the device is ready for use.
+        # This assumes that the pairing skill is loaded as a priority skill
+        # before the rest of the skills are loaded.
+        if not is_paired():
+            self.add_event("mycroft.ready", self.handle_mycroft_ready)
+
+        platform = self.config_core['enclosure'].get('platform', 'unknown')
+        if platform in PLATFORMS_WITH_BUTTON:
+            self.paired_dialog = 'pairing.paired'
```

Comment:
all instance attributes should be defined in __init__().  it can be set to none or to a default value.

Reply:
Good catch, I attempted to do this on line 51 but between that line and this line I changed my mind of the attribute name it seems :)